test/Validate.Capability.cpp

```diff
@@ -40,18 +40,51 @@
 namespace {
 
 using spvtest::ScopedContext;
+using std::get;
 using std::make_pair;
 using std::pair;
 using std::string;
-using std::stringstream;
+using std::ostringstream;
 using std::tuple;
 using std::vector;
 using testing::Combine;
 using testing::Values;
 using testing::ValuesIn;
 
-using ValidateCapability =
-    spvtest::ValidateBase<tuple<string, pair<string, vector<string>>>>;
+// Parameter for validation test fixtures. It's a tuple so Combine() can be used
+// to produce it.  The first string is a capability name, the second an assembly
+// text.  See below methods for how they're interpreted.
+using CapTestParameter = tuple<string, pair<string, vector<string>>>;
+
+const string& Capability(const CapTestParameter& p) { return get<0>(p); }
```

Comment:
Instead of defining `CapTestParameter` as a `tuple` and have multiple standalone functions specifically for it, why not have a `struct` or `class` to group all the data and methods together? I'd think the latter is more easier to read and maintain.